app/src/main/java/com/github/se/eduverse/MainActivity.kt

```diff
@@ -28,33 +30,40 @@ import com.github.se.eduverse.ui.others.OthersScreen
 import com.github.se.eduverse.ui.theme.EduverseTheme
 import com.github.se.eduverse.ui.videos.VideosScreen
 import com.github.se.eduverse.viewmodel.DashboardViewModel
+import com.github.se.eduverse.viewmodel.PhotoViewModel
+import com.github.se.eduverse.viewmodel.PhotoViewModelFactory
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.storage.FirebaseStorage
 import java.io.File
 
 class MainActivity : ComponentActivity() {
 
   private lateinit var auth: FirebaseAuth
   private var cameraPermissionGranted by mutableStateOf(false)
+  private lateinit var photoViewModel: PhotoViewModel
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
 
     // Initialiser Firebase Auth
     auth = FirebaseAuth.getInstance()
-    auth.currentUser?.let {
-      // Déconnexion de l'utilisateur si déjà connecté
+    if (auth.currentUser != null) {
       auth.signOut()
     }
 
-    // Lanceur de demande de permission
+    // Instanciez le repository et le ViewModel
+    val photoRepository =
+        PhotoRepository(FirebaseFirestore.getInstance(), FirebaseStorage.getInstance())
+    val photoViewModelFactory = PhotoViewModelFactory(photoRepository)
+    photoViewModel = ViewModelProvider(this, photoViewModelFactory)[PhotoViewModel::class.java]
+
```

Comment:
Initiate those in the EduverseApp, instead of doing it in MainActivity and the passing the view model to EduverseApp